### ISSUE TITLE
SteamVR 2.0.10 - Dashboard thumbnail not shown for overlay; shows question mark instead

### ISSUE THREAD

**Issue by Raivr-dev** [ValveSoftware/openvr#1785]
SteamVR 2.0.10.

After creating a dashboard overlay using OpenVR's CreateDashboardOverlay(), I then set the thumbnail using SetOverlayFromFile() or SetOverlayRaw().
Instead of showing the image, SteamVR 2.0 shows a question mark.

The SteamVR Overlay Viewer shows the image has been set correctly. The Overlay Viewer does show the correct image, but the SteamVR dashboard shows an image of a question mark.
You can test and reproduce the bug using the Reality Mixer app (AppID 1844610).

An older SteamVR, version 1.27.5, shows the thumbnail correctly. Tested with beta 'temp_1.27.5 - Temporary branch version 1.27.5 (for testing)'.

**Comment by TheDeveloperGuy** [comment]
@kisak-valve 

> Hello @TheDeveloperGuy, this issue is already being tracked at #1785.
> 
> Closing as a duplicate.

It has a very direct impact on perceived quality. Why is a fix "not planned"? What is the use case for "VREvent_DashboardThumbChanged"? What is the workaround in current SteamVR release (as the SteamVR icons show fine)?

**Comment by TheDeveloperGuy** [comment]
Latest SteamVR beta still doesn't address this issue. Can we please fix this regression? Your product bugs are affecting perception of my product. @jeremyselan @aleiby

**Comment by 2601677867** [comment]
This issue still seems to exist in the latest version of SteamVR, but the Reality Mixer app (AppID 1844610) seems to use other methods to solve the problem.

The last time I used iVry, I found that the icons became normal. Could you share how to fix this problem? @TheDeveloperGuy

**Comment by TheDeveloperGuy** [comment]
This is still not fixed, which is par for the course with OpenVR. Report a bug in every forum they have available for the purpose, and they'll ignore it for years. Kind of make reporting bugs redundant. In this instance there seems to be two modes the dashboard works in. In 1 mode the icon is displayed correctly, in the other it isn't.

**Comment by TheDeveloperGuy** [comment]
@kisak-valve Rather than closing bug reports that aren't resolved, maybe getting them assigned is the better path of action?

**Comment by 2601677867** [comment]
> This is still not fixed, which is par for the course with OpenVR. Report a bug in every forum they have available for the purpose, and they'll ignore it for years. Kind of makes reporting bugs redundant. In this instance there seems to be two modes the dashboard works in. In 1 mode the icon is displayed correctly, in the other it isn't.

Thanks for your reply. Do you have any information or code about mode 1 ？

**Comment by TheDeveloperGuy** [comment]
No. You can't switch modes yourself. I guess it's some kind of backward compatibility mode that's used for something. It doesn't really matter, the majority of the time it displays the incorrect icon. I have seen titles display the correct icon, but haven't been able to figure out what they're doing differently.

**Comment by 2601677867** [comment]
> Hi, I'm an fpsVR developer, I don't have this problem. But I'm using SDK version 1.26.7. Maybe this is the reason? On which SDK versions are you seeing this issue?

Hello, the latest version 2.2.3 has this problem, I will try to use version 1.26.7 later. thanks for your advice

**Comment by TheDeveloperGuy** [comment]
> Hi, I'm an fpsVR developer, I don't have this problem. But I'm using SDK version 1.26.7. Maybe this is the reason? On which SDK versions are you seeing this issue?

I've seen it on new and old SDK versions. It's the runtime version that changes it. I think it's related to the image format and size used for the thumbnail. Can you share what size and format your thumbnail is?

**Comment by TheDeveloperGuy** [comment]
Are you using Qt for your dashboard app? 

I'm using the following code to set the thumbnail:

`            vr::Texture_t thumbnail = {(void*)(uint64_t)unThumbnail, vr::TextureType_OpenGL, vr::ColorSpace_Auto };
            vr::VROverlay()->SetOverlayTexture( m_ulOverlayThumbnailHandle, &thumbnail );
`

**Comment by TheDeveloperGuy** [comment]
Thanks, that seems to confirm that the C++ API is broken. The other app I referred to previously is also C#.

**Comment by TheDeveloperGuy** [comment]
Thumbnail handle is valid, tried SetOverlayFromFile() with JPEG and PNG, tried SetOverlayTexture() with GL texture. Tried different sizes. Those methods are definitely being called, and return vr::VROverlayError_None. 

Same code has worked on countless previous SteamVR versions, and works (sometimes see above on SteamVR 2.0) and always on 1.26.7. 

Also, their example doesn't work. This is broken, no doubt about it.

**Comment by TheDeveloperGuy** [comment]
Also doesn't work when compiling with OpenVR 1.26.7 SDK.

**Comment by TheDeveloperGuy** [comment]
OpenVR developer support has always been terrible. But this is a new low. Seems that it doesn't matter that you have a commercial headset driver that brings Valve lots of income, they will still ignore you. Of course, the poor quality of SteamVR reflects on those making drivers for it, not Valve, so what do they care?

**Comment by TheDeveloperGuy** [comment]
To make 100% sure I'm not doing anything incorrectly, I used SetOverlayTexture() to set my icon texture onto the overlay. That works, so the issue is 100% something to do with not being able to set the overlay texture for the thumbnail. Also tried using other textures for setting the thumbnail. Always the same result (ie. a ?).

**Comment by 2601677867** [comment]
> To make 100% sure I'm not doing anything incorrectly, I used SetOverlayTexture() to set my icon texture onto the overlay. That works, so the issue is 100% something to do with not being able to set the overlay texture for the thumbnail. Also tried using other textures for setting the thumbnail. Always the same result (ie. a ?).

The C++ API does seem to be broken, using C# and WPF without any problems.

Also I noticed that the ivry app has fixed this bug, are you the developer of ivry? Or do you use Dashboard Overlay rewritten in other languages to solve this problem?

**Comment by TheDeveloperGuy** [comment]
> 
> Also I noticed that the ivry app has fixed this bug, are you the developer of ivry? Or do you use Dashboard Overlay rewritten in other languages to solve this problem?

Please read my previous reply. It is NOT fixed, otherwise I would not be having this conversation.

**Comment by TheDeveloperGuy** [comment]
RESOLVED: The change in behaviour that was introduced with SteamVR 2.0 is:

Overlay key cannot include spaces.

This is what is causing the icon behaviour. Previous versions of SteamVR didn't have this behaviour. Everything else works correctly. Fix (in lieu of SteamVR fix) is to not include spaces in overlay key.
